src/pages/[confirmAppt].tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from "react";
+import Navbar from "@/components/public/Navbar";
+import { Typography, Box, Link, Button, Grid } from "@mui/material";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import AppointmentParticipantCard from "@/components/appointment/AppointmentParticipantCard";
+import { AppointmentDetail } from "@/types/Appointment";
+import { GetAppointmentsByAppointmentId } from "@/services/Appointments";
+import { useRouter } from "next/router";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { Database } from "supabase/db_types";
+import Loading from "@/components/public/Loading";
+import ConfirmApptDialog from "@/components/appointment/ConfirmApptDialog";
+import { PagePaths } from "enum/pages";
+
+export default function Home() {
+  const router = useRouter();
+  const supabaseClient = useSupabaseClient<Database>();
+  const [appointment, setAppointment] = useState<AppointmentDetail | null>();
+
+  const appointmentId = parseInt(router.query.appointment_id as string);
+  useEffect(() => {
+    console.log(appointmentId);
+    GetAppointmentsByAppointmentId(appointmentId, supabaseClient)
+      .then((appointment) => {
+        console.log(appointment);
+        setAppointment(appointment);
+      })
+      .catch((err) => {
+        console.log(err);
+        return;
+      });
+  }, [supabaseClient, appointmentId]);
+
+  const [choice, setChoice] = useState<"accept" | "reject" | null>(null);
+
+  if (!appointment) return <Loading />;
```

Comment:
Guard is not enough

Reply:
More guard in pages/confirmAppt/[appointment_id].tsx .